lib/modelManagers/reportModel.js

```diff
@@ -0,0 +1,105 @@
+const {
+  Reports,
+  User,
+  Articles,
+  ReportTag
+} = require('../../database/models');
+
+/**
+ * @description handle reporting articles
+ *
+ * @class ReportModelManager
+ */
+class ReportModelManager {
+/**
+ * @description create a new report
+ *
+ * @static
+ * @param {any} reportData Report data to be created
+ * @returns {null} no return
+ *
+ * @memberOf ReportModelManager
+ */
+  static async createReport(reportData) {
+    const createdReport = await Reports.create({
+      userId: reportData.userId,
+      articleId: reportData.articleId,
+      reportTagId: reportData.reportTagId,
+      message: reportData.message
+    });
+    return createdReport;
+  }
+
+  /**
+ * @description get all reports
+ *
+ * @static
+ * @param {any} limit pagination limit
+ * @param {any} offset pagination offset
+ * @returns {Object} return all reports
+ *
+ * @memberOf ReportModelManager
+ */
+  static async getAllReports(limit, offset) {
+    const getAllReports = await Reports.findAll({
+      limit,
+      offset,
+      include: [
+        {
+          model: User,
+          as: 'reporter',
+          attributes: ['firstname', 'lastname']
+        },
+        {
+          model: Articles,
+          as: 'userArticle',
+          attributes: ['title']
+        },
```

Comment:
I don't think including the article is necessary. The article id alone would be fine here

Reply:
Fixed